compiler/codegen/OMRCodeGenerator.cpp

```diff
@@ -3038,10 +3039,20 @@ TR::Instruction *OMR::CodeGenerator::generateDebugCounter(TR::Instruction *curso
       return cursor;
    if (delta == 0)
       return cursor;
-   TR::DebugCounterAggregation *aggregatedCounters = self()->comp()->getPersistentInfo()->getDynamicCounters()->createAggregation(self()->comp());
+   TR::DebugCounterAggregation *aggregatedCounters = self()->comp()->getPersistentInfo()->getDynamicCounters()->createAggregation(self()->comp(), name);
    aggregatedCounters->aggregateStandardCounters(self()->comp(), cursor->getNode(), name, delta, fidelity, staticDelta);
    if (!aggregatedCounters->hasAnyCounters())
       return cursor;
+
+   if (self()->comp()->compileRelocatableCode() && !aggregatedCounters->getReloData(self()->comp(), delta, fidelity, staticDelta))
```

Comment:
you're going to hate me, but I would prefer you create a new query here corresponding to debug counters need to be relocated that itself can just return `compileRelocatableCode()` . That will make it easier to identify this set of code changes later on.